### PR TITLE
Add API module documentation

### PR DESCRIPTION
**Pull request by jonathansamines**

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# `new Heavy(options)`
```
Comment (devinivy) [inline API.md:1]:
```suggestion
## Methods
### `new Heavy(options)`
```

File: API.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+- `options`
+  - `sampleInterval` - frequency of load sampling in milliseconds. Defaults to `0` (no sampling).
+  - `maxHeapUsedBytes` - maximum V8 heap size bytes. Defaults to `0` (no limit).
+  - `maxRssBytes` - maximum process RSS size bytes. Defaults to `0` (no limit).
+  - `maxEventLoopDelay` - maximum event loop delay duration in milliseconds. Defaults to `0` (no limit).
+
+Returns a new `Heavy` object.
+
+## `heavy.start()`
```
Comment (devinivy) [inline API.md:13]:
```suggestion
### `heavy.start()`
```

File: API.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+Starts the sampling operation with `sampleInterval` frequency. When `sampleInterval` is `0`, this operation does nothing.
+
+## `heavy.stop()`
```
Comment (devinivy) [inline API.md:17]:
```suggestion
### `heavy.stop()`
```

File: API.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+Stops the sampling operation.
+
+## `heavy.check()`
```
Comment (devinivy) [inline API.md:21]:
```suggestion
### `heavy.check()`
```

File: API.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+Verifies the current process load and throws a [server unavailable](https://hapi.dev/module/boom/api?#boomserverunavailablemessage-data)
+error when any of the configured limits is exceeded. The current process load is assigned to `error.data`.
+When `sampleInterval` is `0`, this operation does nothing.
+
+## `heavy.load`
```
Comment (devinivy) [inline API.md:27]:
```suggestion
### `heavy.load`
```

**Review by devinivy — APPROVED** [review]
Fresh docs, this is great!  I made some minor suggestions so that the docs will match the format of other simple API docs such as:
 - https://github.com/hapijs/topo/blob/master/API.md
 - https://github.com/hapijs/cryptiles/blob/master/API.md
 - https://github.com/hapijs/iron/blob/master/API.md

I realize that not all the modules' doc formatting are consistent with each other, though, so if you saw this style somewhere else, let's just merge it as-is 👍

**Comment by jonathansamines** [comment]
I had not realized not all modules used the same format. I was borrowing from Podium. Changes applied @devinivy!

**Comment by devinivy** [comment]
The doc formatting could be more consistent across modules— I hadn't noticed podium was a little different than some of these others 🤦 Anyways, these docs are great to have 🥂

**Pull request merged**